xbmc/addons/AddonManager.cpp

```diff
@@ -479,7 +449,61 @@ bool CAddonMgr::HasAvailableUpdates()
   return !GetAvailableUpdates().empty();
 }
 
-bool CAddonMgr::GetAddons(const TYPE &type, VECADDONS &addons, bool enabled /* = true */)
+bool CAddonMgr::GetAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, true);
+}
+
+bool CAddonMgr::GetAddons(VECADDONS& addons, const TYPE& type)
+{
+  return GetAddonsInternal(type, addons, true);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, false);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons, const TYPE& type)
```

Comment:
That's what I was referring to. The default value is unclear (because it's always visible to the callers) and leaks information in this case. It can be used, but why? Overloading does exactly the same, except for those two things. It's also easier to change that ADDON_UNKNOWN magic later, because it's not part of every call anymore when using overloads. Now it's an implementation detail.